Lib/asyncio/staggered.py

```diff
@@ -82,7 +82,11 @@ async def run_one_coro(this_index, coro_fn, this_failed):
             raise _Done
 
     try:
-        async with taskgroups.TaskGroup() as tg:
+        tg = taskgroups.TaskGroup()
+        # Intentionally override the loop in the TaskGroup to avoid
+        # using the running loop, preserving backwards compatibility
+        tg._loop = loop
```

Comment:
This would be wrong if loop is None.

Also how sure are you that TaskGroup doesn’t use its _loop before you get to this point?

Reply:
> This would be wrong if loop is None.

Actually, even when it is None it will work as by default Taskgroup sets _loop to None.

Reply:
> Also how sure are you that TaskGroup doesn’t use its _loop before you get to this point?

Taskgroup only starts using the loop after __aenter__

https://github.com/python/cpython/blob/0881e2d3b1212d988733f1d3acca4011ce5e6280/Lib/asyncio/taskgroups.py#L54-L60

See lines 58-59

Reply:
Please add that to the comment.

Reply:
done